server/middleware/quota_manager.js

```diff
@@ -0,0 +1,131 @@
+var MiniEventEmitter = require('miniee'),
+    QuotaLimiter = require('./quota_limiter.js'),
+    Client = require('../../client/client.js'),
+    logging = require('minilog')('radar:quota_manager');
+
+var QuotaManager = function() { 
+  this._limiters = {};
+};
+
+MiniEventEmitter.mixin(QuotaManager);
+
+QuotaManager.prototype.checkLimits = function(socket, message, messageType, next) {
+  var limiter = this.getLimiter(messageType);
+
+  if (!limiter || (message.op !== 'subscribe' && message.op !== 'sync')) {
+    next();
+  } else if (limiter.isAboveLimit(socket.id)) {
+    logging.warn('#socket.message - rate_limited', message, socket.id);
+
+    socket.send({
+      op: 'err',
+      value: 'rate limited',
+      origin: message
+    });
+
+    next('limit reached');
```

Comment:
next(new Error('limit reached')) for stack traceability - callback errors should always be instanceof Error